### PR TITLE
MOSIP-32453 : Increasing the code coverage for admin service

### PR DESCRIPTION
**Pull request by GOKULRAJ136**

### CODE REVIEW

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/RegistrationCenterServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        registrationCenter.setId(String.valueOf(1L));
+        registrationCenter.setZoneCode("WB-KOL-E");
+        RegCenterPutReqDto dto = new RegCenterPutReqDto();
+        dto.setId(registrationCenter.getId());
+        dto.setZoneCode("WB-MUM-W");
+        List<Device> deviceList = Collections.emptyList();
+        
+        lenient().when(deviceRepository.findByRegIdAndIsDeletedFalseOrIsDeletedIsNull(dto.getId())).thenReturn(deviceList);
+
+        ReflectionTestUtils.invokeMethod(registrationCenterServiceImpl, "validateZoneMachineDevice", registrationCenter, dto);
```
Comment (dhanendra06) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/RegistrationCenterServiceImplTest.java:126]: please add assertion here

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java [hunk 2]
```diff
@@ -160,15 +160,15 @@ public void getAllDefinitionsTest() throws JSONException {
 	
 	@Test
 	@WithUserDetails("reg-officer")
-	public void testFetchAllDynamicFields() throws Exception {		
+	public void testFetchAllDynamicFields() {
```
Comment (dhanendra06) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java:163]: please follow the naming convention for test name

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/utils/RegistrationCenterValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        reqDto.setCenterStartTime(LocalTime.of(8, 0, 0));
+        reqDto.setCenterEndTime(LocalTime.of(18, 0, 0));
+        reqDto.setLunchStartTime(LocalTime.of(12, 30, 0));
+        reqDto.setLunchEndTime(LocalTime.of(13, 30, 0));
+        reqDto.setTimeZone("Asia/Kolkata");
+        reqDto.setZoneCode("zoneCode123");
+        reqDto.setWorkingNonWorkingDays(Collections.singletonMap("MONDAY", true));
+        reqDto.setExceptionalHolidayPutPostDto(Collections.emptyList());
+
+        registrationCenterValidator.createRegCenterPut(newregistrationCenterList, registrationCenter, reqDto);
```
Comment (dhanendra06) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/utils/RegistrationCenterValidatorTest.java:60]: please add assertion here

**Review by dhanendra06 — CHANGES_REQUESTED** [review]

File: admin/admin-service/src/test/java/io/mosip/admin/bulkdataupload/batch/RepositoryListItemWriterTest.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        writer.setOperation("insert");
+
+        entity1 = new T();
+        entity2 = new T();
+    }
+
+    @Test
+    public void testWrite_EmptyChunk() throws Exception {
+        Chunk<T> emptyChunk = new Chunk<>(new ArrayList<>());
+        writer.write(emptyChunk);
```
[the remark this answers is not in the document]
Comment (GOKULRAJ136) [reply admin/admin-service/src/test/java/io/mosip/admin/bulkdataupload/batch/RepositoryListItemWriterTest.java:68]: Added assertions.

File: admin/admin-service/src/test/java/io/mosip/admin/bulkdataupload/batch/RepositoryListItemWriterTest.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    @Test
+    public void testWrite_InsertSuccess() {
+        List<T> entities = new ArrayList<>();
+        entities.add(entity1);
+        entities.add(entity2);
+
```
[the remark this answers is not in the document]
Comment (GOKULRAJ136) [reply admin/admin-service/src/test/java/io/mosip/admin/bulkdataupload/batch/RepositoryListItemWriterTest.java:79]: Added assertions.

File: admin/admin-service/src/test/java/io/mosip/admin/controller/test/AdminControllerTest.java [hunk 6]
```diff
@@ -126,11 +140,169 @@ public void t003lostRidDetailsTest() throws Exception {
...
+		byte[] inputBytes = "sample image data".getBytes();
+
+		ConvertRequestDto convertRequestDto = new ConvertRequestDto();
+		convertRequestDto.setVersion("1.0");
+		convertRequestDto.setModality("Modality");
+		convertRequestDto.setBiometricSubType("Type");
+		convertRequestDto.setPurpose("Action");
+		convertRequestDto.setOnlyImageInformation(0);
+		convertRequestDto.setCompressionRatio(95);
+
```
[the remark this answers is not in the document]
Comment (GOKULRAJ136) [reply admin/admin-service/src/test/java/io/mosip/admin/controller/test/AdminControllerTest.java:161]: Added assertions.

File: admin/hotlist-service/src/test/java/io/mosip/hotlist/security/HotlistSecurityManagerTest.java [hunk 7]
```diff
@@ -58,8 +55,7 @@ public void testHash() {
 	}
 
 	@Test
-	public void testEncrypt()
-			throws JsonParseException, JsonMappingException, JsonProcessingException, IOException, HotlistAppException {
+	public void testEncrypt() throws IOException, HotlistAppException {
```
[the remark this answers is not in the document]
Comment (GOKULRAJ136) [reply admin/hotlist-service/src/test/java/io/mosip/hotlist/security/HotlistSecurityManagerTest.java]: Gave the correct naming convention for the testMethod

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/controller/IntegratedRepositoryTest.java [hunk 8]
```diff
@@ -2312,15 +2242,14 @@ public void t027updateLocationHierarchyDetailsTest() throws Exception {
 						.content(mapper.writeValueAsString(locationRequestDto))).andReturn(),
 				"KER-MSD-097");
 	}
-	@Ignore
+
 	@Test
 	@WithUserDetails("reg-processor")
 	public void tst003getUsersTest2() throws Exception {
```
[the remark this answers is not in the document]
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/controller/IntegratedRepositoryTest.java]: Gave the correct naming convention for the testMethod

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/RegistrationCenterServiceImplTest.java (hunk 1 above)
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/RegistrationCenterServiceImplTest.java:131]: Added assertions.

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java (hunk 2 above)
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java]: Gave the correct naming convention for the testMethod

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/utils/RegistrationCenterValidatorTest.java (hunk 3 above)
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/utils/RegistrationCenterValidatorTest.java:68]: Added assertions.

**Review by dhanendra06 — APPROVED** [review]

**Review by dhanendra06 — APPROVED** [review]

**Pull request merged**